src/write_gcm.c

```diff
@@ -597,26 +591,21 @@ static int wg_curl_get_or_post(char *response_buffer,
   return result;
 }
 
-static size_t wg_curl_write_callback(char *ptr, size_t size, size_t nmemb,
+static size_t wg_curl_write_callback(void *ptr, size_t size, size_t nmemb,
                                      void *userdata) {
-  wg_curl_write_ctx_t *ctx = userdata;
-  if (ctx->size == 0) {
+  size_t requested_bytes = size * nmemb;
+  wg_curl_write_ctx_t *ctx = (wg_curl_write_ctx_t *) userdata;
+
+  ctx->data = realloc(ctx->data, ctx->size + requested_bytes + 1);
```

Comment:
Also this is a memory leak. If realloc fails (returning NULL), you will overwrite your pointer to your old memory block and lose the opportunity to ever free it.

Reply:
Ack, updated.